hw/ip/entropy_src/doc/theory_of_operation.md

```diff
@@ -16,7 +16,13 @@ When entropy is delivered to the downstream hardware block, a signal will indica
 Once the initial boot-time mode phase has completed, the ENTROPY_SRC block can be switched to FIPS/CC compliant mode (for simplicity referred to as FIPS mode) by setting the `FIPS_ENABLE` field in the [`CONF`](registers.md#conf) register to `kMultiBitBool4True`.
 In this mode, once the raw entropy has been health checked, it will be passed into a conditioner block.
 This block will compress the bits such that the entropy bits/physical bits, or min-entropy value, should be improved over the raw data source min-entropy value.
-The compression operation, by default, will compress every 2048 tested bits into 384 full-entropy bits.
+The compression operation will compress every [`HEALTH_TEST_WINDOWS.FIPS_WINDOW`](registers.md#health_test_windows--fips_window) x 4 tested bits into 384 full-entropy bits.
+By default, 2048 tested bits are used.
+Note that after enabling, twice the number of tested bits are used for producing the first seed.
```

Comment:
Maybe add a note about the startup health-test rationale of this. Also note that for a single channel, this is not needed as in the single channel case, if we ever decide to support it, both startup and continuous testing needs 1024 bits.

Reply:
nit: would it be accurate to state at the end:
"Note that after enabling, 1024 samples of 4 bit need to be used for startup health-testing. They are used to produce the first seeds, i.e. 4096 bits ore twice as much are used for this first seed.

Reply:
Excellent, thanks for raising this. I wasn't aware of the 1k samples requirement for the startup health tests. I've now changed the description as suggested but I think we've uncovered an issue here:

The startup health tests comprise always 2 windows independent of the set window size. This means if a window size below 512 samples is configured, the startup test is actually not valid. One could maybe change the window size after performing the startup tests. This should work but DV currently doesn't test this, from `entropy_src_core.sv`:
```sv
  // This code assumes that `health_test_window` does not change dynamically; capture that in an
  // assertion ensuring it only changes when entropy_src is not enabled.
  `ASSERT(HealthTestWindowStableWhenEnabled_A,
          mubi4_test_true_strict(mubi_es_enable) |-> $stable(health_test_window))
```
I think we should at least document this. @zi-v do you think it's realistic that one would like to use 256 samples as window size? I've seen our DV testing it based on my understanding one would rather want to increase the window size to improve the amount of entropy in the seed.

Reply:
Per NIST SP 800-90B table 2, the adaptive proportion test requires 1024 samples for 'Binary data' (which is our 1-bit case) and 512 samples for 'Non-binary data' (more than two distinct values per sample - which is our 4-bit case). 
In addition per section '4.3 Requirements for Health Tests', subsection 4: "The entropy source's startup tests shall run the continuous health tests over at least 1024 consecutive samples"
To me it sounds that the current implementation meets the requirement, even if we run with 1-bit, no harm to still run it twice except for latency. So, using 256 samples window does not seem to comply with 90B unless you can somehow prove that its coverage encapsulates the 512 window case.

Reply:
Thanks for your feedback @zi-v , this is very clear! I think I'll best update the documentation saying that with HT window sizes below 512 samples, one might not meet the 90B requirements.